cmd/resolve.go

```diff
@@ -0,0 +1,596 @@
+/*
+ * Copyright 2018-2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/nats-io/jwt"
+	"github.com/nats-io/nsc/cmd/store"
+
+	"github.com/spf13/cobra"
+)
+
+func createResolveCmd() *cobra.Command {
+	var params ResolveCmdParams
+	var cmd = &cobra.Command{
+		Use:     "resolve",
+		Short:   "Resolve an nsc URL into a JSON blob that can be used by tooling",
+		Example: "resolve nsc://<operator>/?<account>/?<user>",
+		Args:    MaxArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			us := args[0]
+			u, err := ParseNscURL(us)
+			if err != nil {
+				return fmt.Errorf("error parsing %q:%v", us, err)
+			}
+			params.nscu = u
+			q := u.query()
+			if len(q) > 0 {
+				config := GetConfig()
+				oldSR := config.StoreRoot
+				oldOp := config.Operator
+				oldAc := config.Account
+				v, ok := q["store"]
+				if ok {
+					defer func() {
+						config.setStoreRoot(oldSR)
+						if oldOp != "" {
+							config.SetOperator(oldOp)
+						}
+						if oldAc != "" {
+							config.SetAccount(oldAc)
+						}
+					}()
+					sr, err := Expand(v)
+					if err != nil {
+						return err
+					}
+					config.setStoreRoot(sr)
+					config.SetOperator(u.operator)
+				}
+
+				storeDir, ok := q["keystore"]
+				if ok {
+					ks, err := Expand(storeDir)
+					if err != nil {
+						return err
+					}
+					os.Setenv(store.NKeysPathEnv, ks)
+				}
+			}
+			return RunAction(cmd, args, &params)
+		},
+		Hidden: true,
+	}
+	cmd.Flags().StringVarP(&params.outputFile, "output-file", "o", "--", "output file, '--' is stdout")
+
+	return cmd
+}
+
+func init() {
+	GetRootCmd().AddCommand(createResolveCmd())
+}
+
+type ResolveCmdParams struct {
+	nscu       *NscURL
+	results    *ResolveResults
+	oc         *jwt.OperatorClaims
+	ac         *jwt.AccountClaims
+	uc         *jwt.UserClaims
+	outputFile string
+}
+
+type Details struct {
+	Service []string `json:"service,omitempty"`
+	Name    string   `json:"name,omitempty"`
+	Seed    string   `json:"seed,omitempty"`
+	Key     string   `json:"id,omitempty"`
+}
+
+type ResolveResults struct {
+	UserCreds string   `json:"userCreds,omitempty"`
```

Comment:
`user_creds` surely? :)

Reply:
lol - that snake_casing_bad....

Reply:
Just cos JS people cant make up their mind doesnt make it bad :P

Reply:
lolz